service/src/main/java/uk/gov/hmcts/reform/fpl/service/NoticeOfProceedingsTemplateDataGenerationService.java

```diff
@@ -0,0 +1,96 @@
+public class NoticeOfProceedingsTemplateDataGenerationService
+    extends DocmosisTemplateDataGeneration<DocmosisNoticeOfProceeding> {
+
+    private final HearingBookingService hearingBookingService;
+    private final HmctsCourtLookupConfiguration hmctsCourtLookupConfiguration;
+    private final HearingVenueLookUpService hearingVenueLookUpService;
+    private final CaseDataExtractionService caseDataExtractionService;
+    private final Time time;
+
+    @Override
+    public DocmosisNoticeOfProceeding getTemplateData(CaseData caseData) {
+
+        HearingBooking prioritisedHearingBooking = hearingBookingService
+            .getMostUrgentHearingBooking(caseData.getHearingDetails());
+        HearingVenue hearingVenue = hearingVenueLookUpService.getHearingVenue(prioritisedHearingBooking);
+
+        return DocmosisNoticeOfProceeding.builder()
+            .courtName(getCourtName(caseData.getCaseLocalAuthority()))
+            .familyManCaseNumber(caseData.getFamilyManCaseNumber())
+            .todaysDate(formatLocalDateToString(time.now().toLocalDate(), FormatStyle.LONG))
+            .applicantName(PeopleInCaseHelper.getFirstApplicantName(caseData.getApplicants()))
+            .orderTypes(getOrderTypes(caseData.getOrders()))
+            .childrenNames(getAllChildrenNames(caseData.getAllChildren()))
+            .judgeTitleAndName(JudgeAndLegalAdvisorHelper.formatJudgeTitleAndName(
+                caseData.getNoticeOfProceedings().getJudgeAndLegalAdvisor()))
+            .legalAdvisorName(JudgeAndLegalAdvisorHelper.getLegalAdvisorName(
+                caseData.getNoticeOfProceedings().getJudgeAndLegalAdvisor()))
+            .hearingDate(caseDataExtractionService.getHearingDateIfHearingsOnSameDay(
+                prioritisedHearingBooking)
+                .orElse(""))
+            .hearingVenue(hearingVenueLookUpService.buildHearingVenue(hearingVenue))
+            .preHearingAttendance(caseDataExtractionService.extractPrehearingAttendance(
+                prioritisedHearingBooking))
+            .hearingTime(caseDataExtractionService.getHearingTime(prioritisedHearingBooking))
+            .crest(getCrestData())
+            .courtseal(getCourtSealData())
+            .build();
+    }
+
+    private String getCourtName(String courtName) {
+        return hmctsCourtLookupConfiguration.getCourt(courtName).getName();
+    }
+
+    private String getOrderTypes(Orders orders) {
+        return orders.getOrderType().stream()
+            .map(OrderType::getLabel)
+            .collect(Collectors.joining(", "));
+    }
+
+    private String getAllChildrenNames(List<Element<Child>> children) {
+        String childrenNames = children.stream()
+            .map(Element::getValue)
+            .filter(Objects::nonNull)
+            .map(Child::getParty)
+            .filter(Objects::nonNull)
+            .map(childParty -> (childParty.getFirstName()) + " " + (childParty.getLastName()))
```

Comment:
there should be a `getFullName` function that can be used here

Reply:
done